tests/integration/test_create_token.py

```diff
@@ -0,0 +1,134 @@
+from app.authentication.authenticator import decrypt_token
+from tests.integration.app_context_test_case import AppContextTestCase
+from tests.integration.create_token import (
+    PAYLOAD_V2_BUSINESS,
+    PAYLOAD_V2_SOCIAL,
+    PAYLOAD_V2_SUPPLEMENTARY_DATA,
+)
+from tests.integration.integration_test_case import IntegrationTestCase
+
+
+class TestCreateToken(IntegrationTestCase, AppContextTestCase):
+    """
+    The purpose of this test class is to test the creation of a token (from create_token.py) to ensure
+    metadata in a decrypted token is nested/found in the correct level.
+    """
+
+    def test_payload_from_token(self):
+        test_parameters = [
+            {
+                "schema": "test_metadata_routing.json",
+                "theme": "default",
+                "additional_payload": {
+                    "flag_1": 123,
+                    "period_id": "abc",
+                    "link": "https://example.com",
+                },
+                "payload": PAYLOAD_V2_BUSINESS,
+            },
+            {
+                "schema": "social_demo.json",
+                "theme": "social",
+                "additional_payload": {
+                    "flag_1": True,
+                    "user_id": "abc",
+                    "date": "2016-05-12",
+                },
+                "payload": PAYLOAD_V2_SOCIAL,
+            },
+        ]
+        for value in test_parameters:
+            with self.subTest():
+                additional_payload = value.get("additional_payload")
+                token = self.token_generator.create_token_v2(
+                    schema_name=value.get("schema"),
+                    theme=value.get("theme"),
+                    **additional_payload,
+                )
+                with self.test_app.app_context():
+                    decrypted_token = decrypt_token(token)
+                    self.assertEqual(
```

Comment:
I'm not sure about the | here as it makes the test a bit confusing to follow, I'm not sure if maybe there is a nicer way of checking the expected values? I would also avoid multiple assertions where possible - the additional assertion below checks that the keys exist in the decrypted token but should we not check for an exact match with expected keys and values for the whole survey metadata dict, as there could be extra values we do not expect?

Reply:
I've refactored the test where it is one big test, that checks the content & structure of the token is as expected from `decrypted_token` this will cover both assertions.

Reply:
Think this is a lot clearer now 👍